code/controllers/subsystem/vote.dm

```diff
@@ -176,38 +179,31 @@ SUBSYSTEM_DEF(vote)
 			if("gamemode")
 				choices.Add(config.votable_modes)
 			if("groundmap")
+				if(!lower_admin && SSmapping.groundmap_voted)
+					to_chat(usr, "<span class='warning'>The next ground map has already been selected.</span>")
+					return FALSE
 				var/list/maps = list()
-				for(var/i in config.maplist[GROUND_MAP])
-					var/datum/map_config/VM = config.maplist[GROUND_MAP][i]
+				for(var/map in config.maplist[GROUND_MAP])
+					var/datum/map_config/VM = config.maplist[GROUND_MAP][map]
 					if(!VM.voteweight)
 						continue
-					if(VM.config_max_users || VM.config_min_users)
-						var/players = length(GLOB.clients)
-						if(VM.config_max_users && players > VM.config_max_users)
-							continue
-						if(VM.config_min_users && players < VM.config_min_users)
-							continue
-					maps += i
-
-				choices.Add(maps)
-				if(length(choices) < 2)
-					return FALSE
+					maps += VM.map_name
+					shuffle_inplace(maps)
+				for(var/valid_map in maps)
+					choices.Add(valid_map)
 			if("shipmap")
+				if(!lower_admin && SSmapping.shipmap_voted)
+					to_chat(usr, "<span class='warning'>The next ship map has already been selected.</span>")
+					return FALSE
 				var/list/maps = list()
-				for(var/i in config.maplist[SHIP_MAP])
-					var/datum/map_config/VM = config.maplist[SHIP_MAP][i]
+				for(var/map in config.maplist[SHIP_MAP])
+					var/datum/map_config/VM = config.maplist[SHIP_MAP][map]
```

Comment:
So on local, all the ships are available to vote. Maybe it's different in real serb?

Reply:
config based yes